internal/pkg/api/handleEnroll.go

```diff
@@ -146,6 +158,33 @@ func (et *EnrollerT) _enroll(
 	}
 
 	agentID := u.String()
+	// only delete existing agent if it never checked in
+	if agent.Id != "" && agent.LastCheckin == "" {
+		zlog.Debug().
+			Str("EnrollmentId", enrollmentID).
+			Str("AgentId", agent.Id).
+			Str("APIKeyID", agent.AccessAPIKeyID).
+			Msg("Invalidate old api key and remove existing agent with the same enrollment_id")
+		// invalidate previous api key
+		err := invalidateAPIKey(ctx, zlog, et.bulker, agent.AccessAPIKeyID)
+		if err != nil {
```

Comment:
Do this and the `deleteAgent` call below also need to be guarded in the same way the `FindAgent` call is above with `if !errors.Is(err, dl.ErrNotFound) && !strings.Contains(err.Error(), "no such index")` before returning error?

Are there any expected and non-fatal errors that can happen here? I assume that the agents here are not yet enrolled so there's no way a concurrent operation from the UI could interfere?

Reply:
Following up from https://github.com/elastic/fleet-server/pull/2655/files#r1228834438, does `invalidateAPIKey` need to handle the case where the API key is already invalid?

Reply:
I tried the invalidate API with an already invalidated and non existing API key, and the response was successful, so I think we can repeat it safely on a retry. I can add another test for it.